File: basic/extract variable/python/src/example_1.py
"""
Contexto: Tienes una función en un sistema de compras en línea que calcula el descuento aplicado a un producto. La función recibe el precio original y el porcentaje de descuento como argumentos. Actualmente, la fórmula del descuento se encuentra en línea en el código. Aplica "Extract Variable" para mejorar la legibilidad del código al asignar el valor del descuento a una variable con un nombre descriptivo.
"""

def calcular_descuento_(precio_original, porcentaje_descuento):
    precio_con_descuento = precio_original - (precio_original * (porcentaje_descuento / 100))
    return precio_con_descuento

def calcular_descuento(precio_original, porcentaje_descuento):
    descuento = (precio_original * porcentaje_descuento / 100)
    precio_total = precio_original - descuento
    return precio_total

File: basic/extract variable/javascript/example1.js
/*
Contexto: Estás trabajando en una aplicación de conversión de unidades. Tienes una función que convierte grados Celsius a grados Fahrenheit. Actualmente, la fórmula de conversión se encuentra en línea en el código. Aplica "Extract Variable" para mejorar la legibilidad del código al asignar la fórmula de conversión a una variable con un nombre descriptivo.
*/

function celsiusAFahrenheit(celsius) {
    let fahrenheit = (celsius * 9/5) + 32
    return fahrenheit
}

function nuevoCelsiusAFahrenheit(celsius) {
    const tamanioSobreCelsius = 9/5
    const puntoCongelacionAgua = 32
    let fahrenheit = (celsius * tamanioSobreCelsius)  + puntoCongelacionAgua
    return fahrenheit
}

export {celsiusAFahrenheit, nuevoCelsiusAFahrenheit};
